src/machine/image-dbus.c

```diff
@@ -56,11 +61,37 @@ int bus_image_method_remove(
         if (r == 0)
                 return 1; /* Will call us back */
 
-        r = image_remove(image);
-        if (r < 0)
+        if (pipe2(errno_pipe_fd, O_CLOEXEC|O_NONBLOCK) < 0)
+                return sd_bus_error_set_errnof(error, errno, "Failed to create pipe: %m");
+
```

Comment:
I thought about that, but doing this by passing a function parameter isn't thaaat pretty, because we need access to quite a number of parameters in some cases, that are different for each user of the operation stuff. To make the passing possible we'd have to define a struct each time, stick the data in there, and pull it out again, always casting. To me that sounded more complex thatn just having the pipe2() and the fork() explicitly in place here, and using the fact that the stack frame survives the forking...

I also thought about just wrapping the pipe2() and fork() in a single call, but I didn't like this two much as they two calls have little to do with each other, and hiding fork()s in functions that hence return twice I didn't find pretty either. So I thought to stick with the pipe2()/fork() in place each time.